src/IROperator.h

```diff
@@ -797,6 +797,33 @@ inline Tuple tuple_select(const Expr &c0, const Tuple &v0, const Expr &c1, const
 }
 // @}
 
+/** Oftentimes we want to pack a list of expressions with the same type
+ * into a channel dimension, e.g.,
+ * img(x, y, c) = select(c == 0, 100, // Red
+ *                       c == 1, 50,  // Green
+ *                               25); // Blue
+ * This is tedious when the list is long. The following functions
+ * provide convinent syntax that allow one to write:
+ * img(x, y, c) = select_by_id(c, 100, 50, 25);
+ * or
+ * img(x, y, c) = select_by_id(c, {100, 50, 25});
+ */
+// @{
+inline Expr select_by_id(const Expr &id, const std::vector<Expr> &values) {
+    user_assert(values.size() > 0) << "select_by_id only accepts values with size > 0.\n";
+    Expr result = values.back();
+    for (int i = (int)values.size() - 2; i >= 0; i--) {
+        result = select(id == i, values[i], result);
+    }
+    return result;
+}
+template<typename... Args,
+         typename std::enable_if<Halide::Internal::all_are_convertible<Expr, Args...>::value>::type * = nullptr>
+inline Expr select_by_id(const Expr &id, Args &&... values) {
```

Comment:
IMHO, we should move away from this pattern, in favor of just a vector. For one, that's mostly what the Python API does (no varargs, only lists); for another, in C++11 etc adding the extra {brace} wrapping is trivial, and the runtime overhead of a temporary vector construction is unlikely to be meaningful.

Reply:
Removed.